README.md

```diff
@@ -120,6 +120,7 @@ You can use [jetpack](https://github.com/KidkArolis/jetpack) to try these exampl
   * `mode` - one of `history`, `hash`, `memory`, default is `history`
   * `interceptLinks` - whether to handle `<a>` clicks, default is `true`
   * `qs` - a custom query string parser, object of shape `{ parse, stringify }`
+  * `beforeTransition` - called with `(route, data)`, this will be run before a transition happens, if a promise is returned the router will wait for it to be resolved before transitioning
```

Comment:
I wonder if we should pass it as 2nd arg to `start`, that way it's either

`router.start(onTransition)`
or
`router.start(onTransition, beforeTransition)`

Just thinking out loud.
It's a pity we need to add more API :-/

Reply:
Ah, what if we made it possible for onTransition to return a promise.

Oh wait, why do we need `beforeTransition` at all? Hm, why not do your async business in the `onTransition`?

Reply:
Maybe we don't need extra API after all. In fact, I successfully did async logic in onTransition hook in mission-control.